drivers/timer/timer.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <compiler.h>
+#include <dm.h>
+#include <error.h>
+#include <stddef.h>
+#include <drivers/timer.h>
+#include <drivers/wallclock.h>
+
+/* One second at a reference clock rate of 24MHz. */
+#define TICK_INTERVAL 24000000
+
+static uint64_t last_tick;
+static struct device *timer;
+
+int
+timer_cancel_periodic(work_function fn __unused, void *param __unused)
+{
+	return ENOTSUP;
```

Comment:
Since this PR is not WIP, I am assuming we intend to get this merged in. Can you add a TODO here?

Reply:
No this is not WIP. It's complete. Those functions are implemented in the next commit (which can't be squashed since it has a different author).